redox/processor_neworder.go

```diff
@@ -242,6 +268,143 @@ func (o *newOrderProcessor) handleCreateAccount(ctx context.Context, order model
+			return err
+		}
+
+		// Match the order again, after the account is created
+		match, err := o.matchOrder(ctx, id, params.Order)
+		if err != nil {
+			return err
+		}
+		params.Match = *match
+	}
+	return o.handleEnableSummaryReports(ctx, params)
+}
+
+func (o *newOrderProcessor) createTagsForPatient(ctx context.Context, order models.NewOrder, match clinics.EHRMatchResponse) (*clinics.PatientTagIds, error) {
+	tagNames := o.getTagNamesFromOrder(order, match)
+	if tagNames == nil {
+		return nil, nil
+	}
+
+	existingTags := o.getExistingTags(match.Clinic)
+	for _, tagName := range tagNames {
+		_, exists := existingTags[tagName]
+		if !exists {
+			createTag := clinics.CreatePatientTagJSONRequestBody{
+				Name: tagName,
+			}
+			resp, err := o.clinics.CreatePatientTagWithResponse(ctx, *match.Clinic.Id, createTag)
+			if err != nil {
+				return nil, err
+			}
+			if resp.StatusCode() != http.StatusOK && resp.StatusCode() != http.StatusCreated {
+				return nil, fmt.Errorf("unexpected status code %v when creating tagName %s", resp.StatusCode(), tagName)
+			}
+		}
+	}
+
+	resp, err := o.clinics.GetClinicWithResponse(ctx, *match.Clinic.Id)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %vwhen fetching clinic with id %s", resp.StatusCode(), *match.Clinic.Id)
+	}
+
+	existingTags = o.getExistingTags(*resp.JSON200)
+	patientTagIds := clinics.PatientTagIds{}
+	for _, tagName := range tagNames {
+		patientTag, ok := existingTags[tagName]
+		if !ok {
+			// The tag should have been created. If it was deleted in the meantime returning an error will result in a retry
+			return nil, fmt.Errorf("patient tag doesn't exist")
+		}
+		patientTagIds = append(patientTagIds, *patientTag.Id)
+	}
+
+	return &patientTagIds, nil
+}
+
+func (o *newOrderProcessor) replacePatientTags(ctx context.Context, match clinics.EHRMatchResponse, tagIds *clinics.PatientTagIds) error {
+	if tagIds == nil {
+		return nil
+	}
+
+	patient := (*match.Patients)[0]
+	update := clinics.UpdatePatientJSONRequestBody{
+		Email:                          patient.Email,
+		BirthDate:                      patient.BirthDate,
+		FullName:                       patient.FullName,
+		Mrn:                            patient.Mrn,
+		TargetDevices:                  patient.TargetDevices,
+		Tags:                           tagIds,
+	}
+	resp, err := o.clinics.UpdatePatientWithResponse(ctx, *match.Clinic.Id, *patient.Id, update)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return fmt.Errorf("unexpected status code %v replacing tags for patient %s", resp.StatusCode(), *patient.Id)
+	}
+	return nil
+}
+
+func (o *newOrderProcessor) getPatientTagCodes(match clinics.EHRMatchResponse) map[string]struct{} {
+	result := make(map[string]struct{})
+	if match.Settings.Tags.Codes != nil {
+		for _, code := range *match.Settings.Tags.Codes {
```

Comment:
Here, like previous comments, is an unchecked pointer (`Codes`). This could panic. While highly unlikely, consider if it's worth a check to prevent.

...or is this all behind a HTTP request anyway that will recover from the panic I guess? Maybe my previous comments aren't really necessary either then, as I assume this being nil (at least when you expect it not to be nil) is quite rare. I don't love that, but it's probably the best solution. 

I've updated my previous comments for these kinds of things under the assumption that in the rare event these were to panic, there is a `recover()` somewhere that would prevent the service from terminating due to nil pointer dereference.

Reply:
Line 358 is `if match.Settings.Tags.Codes != nil {` so we are checking if `Codes` is `nil`

Reply:
Wow, how the heck did I miss that?